### PR TITLE
fix issue #4

### LINKED ISSUE #4: don't allow move the folder into itself
This is a bug, should fix it early.

### PR DESCRIPTION
**Pull request by holicc**
Fixes # 
Fix #4 
add some test for service.file

### CODE REVIEW

File: service/file.go [hunk 1]
```diff
@@ -46,3 +48,34 @@ func FileCopy(src *model.Matter, parent string) error {
...
+	//eg: a/ -> a/b/c
+	//eg: b/ -> a/b/c
+	if parent != "" && strings.HasPrefix(parent, file.Parent+file.Name) {
+		return false, errors.New("can not move to itself")
+	}
+	//avoid move to same place
+	if parent == file.Parent {
+		return false, errors.New("file already in this dir")
+	}
+	//avoid dst dir has the same name file
```
Comment (saltbo) [inline service/file.go:71]: you can add the `file.name` into the query condition.

In fact, `service. MatterExist` has implemented

Comment (holicc) [reply service/file.go:71]: yes,i notice `service. MatterExist`,but i think it's not necessary to access db again.

File: service/file.go [hunk 2]
```diff
@@ -46,3 +48,34 @@ func FileCopy(src *model.Matter, parent string) error {
...
+	)
+	//check dir exists
+	if !MatterParentExist(uid, parent) {
+		return false, errors.New("dir does not exists")
+	}
+	//avoid move to itself
+	//eg: a/ -> a/
+	//eg: a/ -> a/b/c
+	//eg: b/ -> a/b/c
+	if parent != "" && strings.HasPrefix(parent, file.Parent+file.Name) {
```
Comment (saltbo) [inline service/file.go:64]: `parent + file.name == file.parent`

Just judge whether the target directory is equal to the original directory, right?

**Comment by saltbo** [comment]
http://jartto.wang/2018/12/11/git-rebase/

File: service/file.go [hunk 3]
```diff
@@ -46,3 +48,34 @@ func FileCopy(src *model.Matter, parent string) error {
...
+		return false, errors.New("can not move to itself")
+	}
+	//avoid move to same place
+	if parent == file.Parent {
+		return false, errors.New("file already in this dir")
+	}
+	//avoid dst dir has the same name file
+	if err := gormutil.DB().Where("uid=? and parent = ?", uid, parent).Find(&out).Error; err != nil {
```
Comment (saltbo) [inline service/file.go:72]: What I said is that the database query and the following traversal can be replaced by `service. MatterExist`

File: service/file.go (hunk 2 above)
Comment (holicc) [reply service/file.go:64]: uh no,this condition not only avoid folder move to itself,but also avoid folder move to it's children folder.

`//eg: a/ -> a/`  means: move folder `a` to itself . so  `parent + file.name == file.parent`  =>  `a/a == ''`

**Review by saltbo — APPROVED** [review]
LGTM

**Pull request merged**